contxt/services/contxt_deployments.py

```diff
@@ -0,0 +1,52 @@
+import os
+from typing import List, Optional
+
+from dotenv import load_dotenv
+from requests.exceptions import HTTPError
+
+from ..auth import Auth
+from ..models.contxt import Cluster
+from .api import ApiEnvironment, ConfiguredApi
+
+load_dotenv()
+
+
+class ContxtDeploymentService(ConfiguredApi):
+    """Contxt API client"""
+
+    _envs = (
+        ApiEnvironment(
+            name="production",
+            base_url="https://contxt.api.ndustrial.io/deploy/v1",
+            client_id="8qY2xJob1JAxhmVhIDLCNnGriTM9bct8",
+        ),
+        ApiEnvironment(
+            name="staging",
+            base_url="https://contxt-api.staging.ndustrial.io/deploy/v1",
+            client_id="qGzdTXcmB57zlTp86rYsivG9qEss1lbF",
+        ),
+    )
+
+    def __init__(self, auth: Auth, env: str = os.getenv("env", "production"), **kwargs) -> None:
+        super().__init__(env=env, auth=auth, **kwargs)
+
+    def get_clusters(self, organization_id: str) -> List[Cluster]:
+        return [Cluster.from_api(rec) for rec in self.get(f"{organization_id}/clusters")]
+
+    def get_cluster(self, organization_id: str, cluster_slug: str) -> Cluster:
+        resp = self.get(f"{organization_id}/clusters/{cluster_slug}")
+        return Cluster.from_api(resp)
+
+    def register_cluster(
+        self, organization_id: str, cluster: Cluster, secret_bearer_token: Optional[str]
+    ) -> None:
+        obj = cluster.post()
+        # for legacy / DC/OS cluster registration, a token should be provided
+        if secret_bearer_token:
+            obj["secrets"] = {"BEARER_TOKEN": secret_bearer_token}
+        try:
+            resp = self.post(f"{organization_id}/clusters", json=obj)
+            return Cluster.from_api(resp)
+        except HTTPError:
+            pass
+            return
```

Comment:
was this try/catch for debugging?

Reply:
No I just wrapped it in this so the error wouldn't bubble up and dump a stack trace in the output because something built in was already printing the actual error log